src/backend/dx11/src/lib.rs

```diff
@@ -1057,69 +1101,137 @@ impl hal::command::RawCommandBuffer<Backend> for CommandBuffer {
-        self.render_pass_cache = None;
-        self.bound_bindings = 0;
-        self.required_bindings = None;
-        self.max_bindings = None;
-        self.vertex_buffers.clear();
-        self.vertex_offsets.clear();
-        self.vertex_strides.clear();
+        self.reset();
     }
 
-    fn begin_render_pass<T>(&mut self, render_pass: &RenderPass, framebuffer: &Framebuffer, _target_rect: pso::Rect, clear_values: T, _first_subpass: command::SubpassContents)
+    fn begin_render_pass<T>(&mut self, render_pass: &RenderPass, framebuffer: &Framebuffer, target_rect: pso::Rect, clear_values: T, _first_subpass: command::SubpassContents)
     where
         T: IntoIterator,
         T::Item: Borrow<command::ClearValueRaw>,
     {
         let mut clear_iter = clear_values.into_iter();
-        let attachment_clears = render_pass.attachments
-            .iter()
-            .enumerate()
-            .map(|(i, attachment)| {
-                let cv = if attachment.ops.load == pass::AttachmentLoadOp::Clear || attachment.stencil_ops.load == pass::AttachmentLoadOp::Clear {
-                    Some(*clear_iter.next().unwrap().borrow())
-                } else {
-                    None
-                };
+        let mut attachment_clears = Vec::new();
+
+        for (idx, attachment) in render_pass.attachments.iter().enumerate() {
+            use pass::AttachmentLoadOp::*;
+
+            //let attachment = render_pass.attachments[attachment_ref];
+            let format = attachment.format.unwrap();
+            let channel_type = format.base_format().1;
+
+            fn typed_clear_color(ty: ChannelType, raw_clear: ClearColorRaw) -> ClearColor {
+
+                match ty {
+                    ChannelType::Unorm | ChannelType::Inorm | ChannelType::Ufloat |
+                    ChannelType::Float | ChannelType::Uscaled | ChannelType::Iscaled |
+                    ChannelType::Srgb => ClearColor::Float(unsafe { raw_clear.float32}),
+
+                    ChannelType::Uint => ClearColor::Uint(unsafe { raw_clear.uint32}),
 
-                AttachmentClear {
-                    subpass_id: render_pass.subpasses.iter().position(|sp| sp.is_using(i)),
-                    value: if attachment.ops.load == pass::AttachmentLoadOp::Clear {
-                        assert!(cv.is_some());
-                        cv
-                    } else {
-                        None
+                    ChannelType::Int => ClearColor::Int(unsafe { raw_clear.int32}),
+                }
+            }
+
+            let subpass_id = render_pass.subpasses.iter().position(|sp| sp.is_using(idx));
+
+            if attachment.ops.load == Clear ||
+               attachment.stencil_ops.load == Clear
+            {
+                let raw_clear_value = *clear_iter.next().unwrap().borrow();
+
+                match (attachment.ops.load, attachment.stencil_ops.load) {
+                    (Clear, Clear) => {
+                        if format.is_depth() {
```

Comment:
let's put the condition inside the match, i.e. `(Clear, Clear) if format.is_depth() => ...`